model/build-processor/src/main/java/org/keycloak/models/map/processor/GenerateEntityImplementationsProcessor.java

```diff
@@ -0,0 +1,327 @@
+//          .sorted(Comparator.comparing(Map.Entry::getKey))
+//          .forEach(me -> processingEnv.getMessager().printMessage(
+//              Diagnostic.Kind.NOTE,
+//              "** " + me.getKey() + ": " + me.getValue().stream().map(ExecutableElement::getSimpleName).sorted(Comparator.comparing(Object::toString)).collect(Collectors.joining(", ")))
+//          );
+    }
+
+    private static final Pattern BEAN_NAME = Pattern.compile("(get|set|is|delete|remove|add|update)([A-Z]\\S+)");
+    private static final Map<String, String> FORBIDDEN_PREFIXES = new HashMap<>();
+    static {
+        FORBIDDEN_PREFIXES.put("delete", "remove");
+    }
+
+    private String determineAttributeFromMethodName(ExecutableElement e) {
+        Name name = e.getSimpleName();
+        Matcher m = BEAN_NAME.matcher(name.toString());
+        if (m.matches()) {
+            String prefix = m.group(1);
+            if (FORBIDDEN_PREFIXES.containsKey(prefix)) {
+                processingEnv.getMessager().printMessage(
+                  Kind.ERROR,
+                  "Forbidden prefix " + prefix + "... detected, use " + FORBIDDEN_PREFIXES.get(prefix) + "... instead", e
+                );
+            }
+            return m.group(2);
+        }
+        return null;
+    }
+
+    private void generateImpl(TypeElement e, Map<String, HashSet<ExecutableElement>> methodsPerAttribute) throws IOException {
+        GenerateEntityImplementations an = e.getAnnotation(GenerateEntityImplementations.class);
+        Elements elements = processingEnv.getElementUtils();
+        TypeElement parentTypeElement = elements.getTypeElement(an.inherits().isEmpty() ? "void" : an.inherits());
+        final List<? extends Element> allMembers = elements.getAllMembers(parentTypeElement);
+        String className = e.getQualifiedName().toString();
+        String packageName = null;
+        int lastDot = className.lastIndexOf('.');
+        if (lastDot > 0) {
+            packageName = className.substring(0, lastDot);
+        }
+
+        String simpleClassName = className.substring(lastDot + 1);
+        String mapImplClassName = className + "Impl";
+        String mapSimpleClassName = simpleClassName + "Impl";
+
+        JavaFileObject enumFile = processingEnv.getFiler().createSourceFile(mapImplClassName);
+        try (PrintWriter pw = new PrintWriter(enumFile.openWriter()) {
+            @Override
+            public void println(String x) {
+                super.println(x == null ? x : x.replaceAll("java.lang.", ""));
+            }
+        }) {
+            if (packageName != null) {
+                pw.println("package " + packageName + ";");
+            }
+
+            pw.println("import java.util.EnumMap;");
+            pw.println("import java.util.Objects;");
+            pw.println("public class " + mapSimpleClassName + (an.inherits().isEmpty() ? "" : " extends " + an.inherits()) + " implements " + className + " {");
```

Comment:
Should we add some warning that the class is generated and should not be changed manually?

Reply:
good idea